lib/src/widgets/overlay.dart

```diff
@@ -143,14 +154,27 @@ class _DescribedFeatureOverlayState extends State<DescribedFeatureOverlay>
     _openController.dispose();
     _completeController.dispose();
     _dismissController.dispose();
-    _pulseController?.dispose();
+    _pulseController.dispose();
     super.dispose();
   }
 
+  @override
+  void didUpdateWidget(DescribedFeatureOverlay oldWidget) {
+    if (oldWidget.enablePulsingAnimation != widget.enablePulsingAnimation) {
+      if (widget.enablePulsingAnimation)
+        _pulseController.forward(from: 0);
+      else {
+        _pulseController.stop();
+        setState(() => _transitionProgress = 0);
+      }
+    }
+    super.didUpdateWidget(oldWidget);
+  }
+
   @override
   void didChangeDependencies() {
     super.didChangeDependencies();
-    final Bloc bloc = FeatureDiscovery._blocOf(context);
+    final Bloc bloc = Provider.of<Bloc>(context, listen: false);
```

Comment:
Why? The previous way to get the bloc should be preferred as it contains an assertion that checks if there's actually a bloc

Reply:
Then we should make it public. As you said, we should not use the `part` directive.

Reply:
Right, since you changed the organization of files.

Reply:
Actually, check my next commit. I simply added a method for that class as well as the other one is private.